.travis.yml

```diff
@@ -1,18 +1,59 @@
 ---
+# Add additional stages in the order of execution here, and then under the job:include: key
+dist: "focal"  # Ubuntu 20.04
+stages:
+  - name: "lint"
+  - name: "build"
+  - name: "test"
+  - name: "integration"
+  # - name: "deploy"
+  #   if: "branch = master"
+
 language: "python"
-services:
-  - "docker"
 python:
   - 3.7
   - 3.8
 
-before_script:
-  - "pip install invoke toml poetry ansible pynetbox"
-  - "ansible-galaxy collection install netbox.netbox"
-  - "poetry install"
-  - "invoke build-image -p $TRAVIS_PYTHON_VERSION"
+services:
+  - "docker"
 
+# Script & Before Script for test stage
+before_script:
+  - "pip install invoke"
+  - "invoke build-image --nocache"
 script:
-  # TODO Update invoke to pass the version of python.
-  - "invoke tests"
-  - "invoke integration-tests"
+  - "invoke pytest"
+
+jobs:
+  include:
+    - stage: "lint"
+      env:
+        - "INVOKE_LOCAL=True"
+      before_script:
+        - "pip install invoke poetry toml"
```

Comment:
are you sure you want to pip install poetry?

Reply:
I thought there was recommendations against this, but it looks like it is valid.

Reply:
As far as I understand it's ok to do that in docker and in closed environment like travis